jb/child_allowance.py

```diff
@@ -150,72 +145,116 @@
 )
 
 # Define a function to calculate poverty rates
-def pov(groupby, data=person_sim):
-    return (
-        data.groupby(groupby)
-        .apply(lambda x: mdf.weighted_mean(x, "poverty_flag", "asecwt"))
-        .reset_index()
+def pov(data, group):
+    return pd.DataFrame(
+        mdf.weighted_mean(data, "poverty_flag", "asecwt", groupby=group)
     )
 
 
-# Function to be simplified with microdf update to:
-# `return mdf.weighted_mean(data, "poverty_flag", "asecwt",groupby)
-# .reset_index()`
-
-# Poverty rate and demographic-based heterogenous poverty rates
-poverty_rate = pov("sim_flag")
-poverty_rate_sex = pov(["sim_flag", "sex"])
-poverty_rate_race_hispan = pov(["sim_flag", "race_hispan"])
-poverty_rate_state = pov(["sim_flag", "state"])
+# Poverty rate and state/demographic-based heterogenous poverty rates
+poverty_rate = pov(person_sim, "sim_flag")
+poverty_rate_state = pov(person_sim, ["sim_flag", "state"])
+poverty_rate_sex = pov(person_sim, ["sim_flag", "sex"])
+poverty_rate_race_hispan = pov(person_sim, ["sim_flag", "race_hispan"])
 
 # Child poverty rate
-poverty_rate_child = pov("sim_flag", person_sim[person_sim.child_6])
+poverty_rate_child = pov(person_sim[person_sim.child_6], "sim_flag")
 
 # Rename constructed poverty_rates
 poverty_rates = [
     poverty_rate,
     poverty_rate_sex,
     poverty_rate_race_hispan,
-    poverty_rate_race_hispan,
     poverty_rate_state,
     poverty_rate_child,
 ]
 for i in poverty_rates:
     i.rename({0: "poverty_rate"}, axis=1, inplace=True)
 
+"""
+The following code creates a pivot table to examine in detail
+the impacts of each policy on state-based outcomes. The procedure
+is replicable for any of the demographics of interest.
+"""
+
+
+# Define percentage change functions
+def pp_change(base, new):
```

Comment:
I'd skip this being a function and just subtract when it's needed, otherwise readers will have to look at the function to see if it's something more complex than it appears to need